host/libhackrf/src/hackrf.h

```diff
@@ -43,88 +43,730 @@ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSI
+ * 
+ * Typical error-handling code example:
+ * ```c
+ *  result = hackrf_init();
+ *  if (result != HACKRF_SUCCESS) {
+ *      fprintf(stderr,
+ *          "hackrf_init() failed: %s (%d)\n",
+ *          hackrf_error_name(result),
+ *          result);
+ *      return EXIT_FAILURE;
+ *  }
+ * ```
+ * 
+ * Instead of `if (result != HACKRF_SUCCESS)` the line `if (result)` can also be used with the exact same behaviour.
+ * 
+ * The special case @ref HACKRF_TRUE is only used by @ref hackrf_is_streaming
+ * 
+ * # Enum conversion
+ * 
+ * Most of the enums defined in libhackrf have a corresponding `_name` function that converts the enum value into a human-readable string. All strings returned by these functions are statically allocated and do not need to be `free`d. An example is the already mentioned @ref hackrf_error_name function for the @ref hackrf_error enum.
+ * 
+ * # Library internals
+ * 
+ * The library uses `libusb` (version 1.0) to communicate with HackRF hardware. It uses both the synchronous and asynchronous API for communication (asynchronous for streaming data to/from the device, and synchronous for everything else). The asynchronous API requires to periodically call a variant of `libusb_handle_events`, so the library creates a new "transfer thread" for each device doing that using the `pthread` library. The library uses multiple transfers for each device (@ref hackrf_get_transfer_queue_depth).
+ */
+
+/**
+ * @defgroup device Device listing, opening, closing and querying
+ * 
+ * @brief managing HackRF devices and querying information about them
+ * 
+ * The libhackrf library interacts via HackRF hardware through a @ref hackrf_device handle. This handle is opaque, meaning its fields are internal to the library and should not be accessed by user code. To use a device, it first needs to be opened, than it can be interacted with, and finally the device needs to be closed via @ref hackrf_close.
+ * 
+ * # Opening devices
+ * 
+ * ## Open first device
+ * 
+ * @ref hackrf_open opens the first USB device (chosen by libusb). Useful if only one HackRF device is expected to be present.
+ * 
+ * ## Open by serial
+ * 
+ * @ref hackrf_open_by_serial opens a device by a given serial (suffix). If no serial is specified it defaults to @ref hackrf_open
+ * 
+ * ## Open by listing
+ * 
+ * All connected HackRF devices can be listed via @ref hackrf_device_list. The list must be freed by @ref hackrf_device_list_free.
+ * 
+ * This struct lists all devices and their serial numbers. Any one of them can be opened by @ref hackrf_device_list_open. All the fields should be treated read-only!
+ * 
+ * # Closing devices
+ * 
+ * If the device is not needed anymore, then it can be closed via @ref hackrf_close. Closing a device terminates all ongoing transfers, and resets the device to IDLE mode.
+ * 
+ * # Querying device information
+ * 
+ * ## board ID
```

Comment:
This and other headings in this section should have their first letters capitalized.